Source/Game/Game/Editor/EaseCurveTool.cpp

```diff
@@ -0,0 +1,402 @@
+
+                    drawList->AddLine(VLTop, VLBottom, IM_COL32(255,255,255,100));
+                    drawList->AddLine(VRTop, VRBottom, IM_COL32(255,255,255,100));
+                    drawList->AddLine(HLeft, HRight, IM_COL32(255,255,255,100));
+
+                    f32 start = p0.x + (size.x * invHalfCanvasScale);
+                    f32 end = p0.x + (size.x * (1.f - invHalfCanvasScale));
+                    if (useInverse && _inverse)
+                    {
+                        start = end;
+                        end = p0.x + (size.x * invHalfCanvasScale);
+                    }
+
+                    f32 x = Spline::Interpolation::Linear::Lerp(_timer / _timerScale, start, end);
+                    f32 y = p0.y + (size.y / 2.0f);
+
+                    drawList->AddCircleFilled(ImVec2(x, y), 8.f, IM_COL32(255,0,0,255));
+                    drawList->AddRect(p0, p1, IM_COL32(255,255,255,255));
+                    drawList->PopClipRect();
+                }
+
+                // ease preview
+                {
+                    ImGui::Text("%s", easeName[selectedEaseName]);
+                    ImVec2 size(320.f, 48.f);
+                    ImGui::InvisibleButton("EaseCanvas", size);
+                    ImVec2 p0 = ImGui::GetItemRectMin();
+                    ImVec2 p1 = ImGui::GetItemRectMax();
+                    ImDrawList* drawList = ImGui::GetWindowDrawList();
+                    drawList->PushClipRect(p0, p1);
+
+                    ImVec2 VLTop = { p0.x + (size.x * invHalfCanvasScale), p0.y };
+                    ImVec2 VLBottom = { VLTop.x, p1.y };
+                    ImVec2 VRTop = { p1.x - (size.x * invHalfCanvasScale), p0.y };
+                    ImVec2 VRBottom = { VRTop.x, p1.y };
+                    ImVec2 HLeft = { p0.x, p1.y - (size.y * 0.5f) };
+                    ImVec2 HRight = { p1.x, HLeft.y };
+
+                    drawList->AddLine(VLTop, VLBottom, IM_COL32(255,255,255,100));
+                    drawList->AddLine(VRTop, VRBottom, IM_COL32(255,255,255,100));
+                    drawList->AddLine(HLeft, HRight, IM_COL32(255,255,255,100));
+
+                    f32 x = GetEasePoint(_timer / _timerScale, selectedEaseType, selectedEaseName, polynomialDegree);
+
+                    f32 xOffset = x * (size.x * canvasScale);
+                    f32 x0 = p0.x + (size.x * invHalfCanvasScale) + xOffset;
+                    if (useInverse && _inverse)
+                        x0 = p0.x + (size.x * (1.f - invHalfCanvasScale)) - xOffset;
+
+                    f32 y0 = p0.y + (size.y / 2.0f);
+
+                    drawList->AddCircleFilled(ImVec2(x0, y0), 8.f, IM_COL32(255,0,0,255));
+
+
+                    drawList->AddRect(p0, p1, IM_COL32(255,255,255,255));
+                    drawList->PopClipRect();
+                }
+            }
+        }
+        ImGui::End();
+    }
+
+    void EaseCurveTool::Update(f32 deltaTime)
+    {
+        _timer += deltaTime;
+        if (_timer > _timerScale)
```

Comment:
Should we be using > over >=?

Should we the _timer to 0.0f here when calling ResetTimer instead of subtracting _timerScale from it? Unless specifically needing to be set to zero, this causes inaccuracies over time